src/random.cpp

```diff
@@ -249,7 +249,22 @@ static void Strengthen(const unsigned char (&seed)[32], int microseconds, CSHA51
     memory_cleanse(buffer, sizeof(buffer));
 }
 
-#ifndef WIN32
+#if defined(WIN32)
```

Comment:
I don't know if I like effectively duplicating the which-random-source-to-use logic in macro statements. It feels like this too risks overlooking things.

One possibility is adding a `(void)GetDevURandom;` line in the exact places where we know that no /dev/urandom is needed (but only there, so that it doesn't silence an unintended lack of invocation). Such a statement will also silence the warning, and since the method is static, if it's unused, it still won't be included in the object code.

Reply:
Ok, another ingenious way to silence this warning. The previous one did not gain traction for a few months, so lets see if this one flies. Updated the patch.

I confirm that the warning is gone and that the symbol is not included in the object code with `-O1` or higher. It is included with `-O0` or in the absence of any `-O...` option but this is fine.